milli/src/update/index_documents/transform.rs

```diff
@@ -347,25 +356,26 @@ impl Transform<'_, '_> {
     /// Generate the `TransformOutput` based on the given sorter that can be generated from any
     /// format like CSV, JSON or JSON stream. This sorter must contain a key that is the document
     /// id for the user side and the value must be an obkv where keys are valid fields ids.
-    fn output_from_sorter<F>(
+    fn output_from_sorter<F, E>(
         self,
-        sorter: grenad::Sorter<MergeFn>,
+        sorter: grenad::Sorter<MergeFn<E>>,
         primary_key: String,
         fields_ids_map: FieldsIdsMap,
         approximate_number_of_documents: usize,
         mut external_documents_ids: ExternalDocumentsIds<'_>,
         progress_callback: F,
-    ) -> anyhow::Result<TransformOutput>
+    ) -> Result<TransformOutput>
     where
         F: Fn(UpdateIndexingStep) + Sync,
+        Error: From<E>,
     {
         let documents_ids = self.index.documents_ids(self.rtxn)?;
         let mut fields_distribution = self.index.fields_distribution(self.rtxn)?;
         let mut available_documents_ids = AvailableDocumentsIds::from_documents_ids(&documents_ids);
 
         // Once we have sort and deduplicated the documents we write them into a final file.
         let mut final_sorter = create_sorter(
-            |_docid, _obkvs| Err(anyhow!("cannot merge two documents")),
+            |_id, _obkvs| Err(InternalError::IndexingMergingKeys { process: "merging documents" }),
```

Comment:
```suggestion
            |_id, _obkvs| Err(InternalError::IndexingMergingKeys { process: "documents" }),
```

wold give `invalid merge while processing {documents}`